astropy/cosmology/_src/traits/darkmatter.py

```diff
@@ -0,0 +1,51 @@
+"""Dark matter component.
+
+This is private API. See `~astropy.cosmology.traits` for public API.
+
+"""
+
+__all__ = ["_DarkMatterComponent"]
```

Comment:
Let's actually make this public API! the question of `inv_efunc` is ongoing, but given its necessity, I don't think it should prevent us from making this (and the other classes) public. 
So this means a changelog is necessary.

```suggestion
__all__ = ["DarkMatterComponent"]
```

Reply:
Alright. Done for making it public. I am starting another PR so we can make the other api's public too.